src/groupBy.test.ts

```diff
@@ -176,6 +176,56 @@ describe("typing", () => {
     >();
   });
 
+  test("symbol", () => {
+    const mySymbol = Symbol("mySymbol");
+    const data = groupBy(
+      [
+        { [mySymbol]: "cat", b: 123 },
+        { [mySymbol]: "dog", b: 456 },
+      ],
+      () => mySymbol,
+    );
+    expectTypeOf(data).toEqualTypeOf<
+      Partial<
+        Record<
+          typeof mySymbol,
+          NonEmptyArray<{ [mySymbol]: string; b: number }>
+        >
+      >
+    >();
+  });
+
+  test("branded string", () => {
+    type Branded<K, T> = K & { _type: T };
+    type UserID = Branded<string, "UserId">;
```

Comment:
I prefer pulling out types to the module level, I don't have anything smart for a reason, it is just always a bit jarring to see a type definition inside a test

Reply:
(test removed)